monitoring/boot_config.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2016 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"syscall"
+
+	"github.com/gravitational/satellite/agent/health"
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+	"github.com/gravitational/trace"
+)
+
+func int8string(bs []int8) string {
+	b := make([]byte, 0, len(bs))
+	for _, v := range bs {
+		if v == 0 {
+			break
+		}
+		b = append(b, byte(v))
+	}
+	return string(b)
+}
+
+var (
+	reBootConfigParam   = regexp.MustCompile(`(\S+)\=([ym])`)
+	reBootConfigComment = regexp.MustCompile(`#.*`)
+)
+
+func parseBootConfig() (map[string]string, error) {
+	var uname syscall.Utsname
+	err := syscall.Uname(&uname)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	file, err := os.Open(fmt.Sprintf("/boot/config-%s", int8string(uname.Release[:])))
+	if err != nil {
+		return nil, trace.ConvertSystemError(err)
+	}
+	defer file.Close()
+
+	params := map[string]string{}
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		if reBootConfigComment.Match(scanner.Bytes()) || scanner.Text() == "" {
+			continue
+		}
+
+		parsed := reBootConfigParam.FindStringSubmatch(scanner.Text())
+		if len(parsed) != 3 {
+			continue
+		}
+
+		params[parsed[1]] = parsed[2]
+	}
+
+	err = scanner.Err()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return params, nil
+}
+
+// BootConfigParam defines parameter name (without CONFIG_ prefix) to check
+type BootConfigParam struct {
+	// Param is boot config parameter to check
+	Name string
+}
+
+// BootConfigParamChecker checks whether parameters provided are specified in linux boot configuration file
+type BootConfigParamChecker struct {
+	// Params is array of parameters to check for
+	Params []BootConfigParam
+}
+
+const (
+	bootConfigParamID = "boot-config"
+)
+
+// Name returns name of the checker
+func (c *BootConfigParamChecker) Name() string {
+	return bootConfigParamID
+}
+
+// Check parses boot config files and validates whether parameters provided are set
+func (c *BootConfigParamChecker) Check(ctx context.Context, reporter health.Reporter) {
+	cfg, err := parseBootConfig()
+	if trace.IsNotFound(err) {
+		reporter.Add(NewProbeFromErr(bootConfigParamID,
```

Comment:
This is not an error, right?

Reply:
that's for a side effect to log the underlying error where you could see filename etc when invoked as part of gravity checks